walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,214 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.TreeSet;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
+  /**
+   * Finds the possible TimeRanges where the mandatory attendees can attend
+   * @param events_list sorted list of events in ascending start time order
+   * @param request includes mandatory attendees and duration of meeting
+   * @return collection of possible TimeRanges the meeting can be scheduled in  
+   */
+  private Collection<TimeRange> intervalFinder(List<Event> events_list, MeetingRequest request) {
+    Collection<String> mandatory = request.getAttendees();
+    long duration = request.getDuration();
+    List<TimeRange> possibleTimes = new ArrayList<>();
+
+    int prev_end = 0;
+      for (Event e : events_list){
+        // Determine if the attendees for the meeting are busy at other events
+        Collection<String> intersection = new HashSet<>(mandatory);
+        intersection.retainAll(e.getAttendees());
+        TimeRange range = e.getWhen();
+        if (intersection.size() > 0) {
+          TimeRange possibleRange = TimeRange.fromStartEnd(prev_end, range.start(), false);
+          if (duration <= possibleRange.duration()) {
+            possibleTimes.add(possibleRange);
+          }
+          if (prev_end < range.end()) {
+            prev_end = range.end();
+          }
+        }
+      }
+      if (prev_end != TimeRange.END_OF_DAY + 1) {
+        possibleTimes.add(TimeRange.fromStartEnd(prev_end, TimeRange.END_OF_DAY, true));
+      }
+      return possibleTimes;
+  }
+
+  /**
+   * Finds a specific attendee's events for the day.
+   * @param events_list sorted list of events in ascending start time order
+   * @param attendee specific attendee to search for
+   * @return collection of events that attendee has on the schedule
+   */
+  private Collection<Event> eventFinderWithAttendee(Collection<Event> events_list, String attendee) {
+    List<Event> eventsByAttendee = new ArrayList<Event>(); 
+
+    for (Event e : events_list){
+      if (e.getAttendees().contains(attendee)){
+        eventsByAttendee.add(e);
+      }
+    }
+    return eventsByAttendee;
+  }
+
+  /**
+   * Split large range into smaller pieces without event overlapping
```

Comment:
Same thing here, maybe explain how you are doing this, because it's hard to trace